src/Build/Instance/TaskRegistry.cs

```diff
@@ -1182,21 +1182,27 @@ internal class Stats()
                 public short ExecutedCount { get; private set; } = 0;
                 public long TotalMemoryConsumption { get; private set; } = 0;
                 private readonly Stopwatch _executedSw  = new Stopwatch();
+#if NET
                 private long _memoryConsumptionOnStart;
+#endif
 
                 public TimeSpan ExecutedTime => _executedSw.Elapsed;
 
                 public void ExecutionStarted()
                 {
-                    _memoryConsumptionOnStart = GC.GetTotalMemory(false);
+#if NET
+                    _memoryConsumptionOnStart = GC.GetTotalAllocatedBytes(false);
```

Comment:
We want this behavior in net472 distribution of MSBuild.exe so please add an `#else` with the previous behavior. 
Though if this is causing perf trouble, it's a perf bug that it runs by default, but that can be solved in subsequent PR by me.